compiler/runtime/runtime64_specific.cpp

```diff
@@ -349,27 +353,32 @@ char* SKIP_open_file(char* filename_obj) {
   char* filename = sk2c_string(filename_obj);
 
   int fd = open(filename, O_RDONLY);
-  int status = fstat(fd, &s);
-  size_t size = s.st_size;
-
-  char* result = nullptr;
-  if (size == 0) {
-    result = sk_string_alloc(0);
-    sk_string_set_hash(result);
-    return result;
+  if (fd == -1) {
+    perror("ERROR (open failed)");
+    fprintf(stderr, "Could not open file: %s\n", filename);
+    exit(ERROR_FILE_IO);
   }
```

Comment:
While I understand that the control flow regarding testing size for 0 in the new version is equivalent, I think the old code organization --- where this special case is handled by a test for the special case followed by an early return --- is easier to read. It lets you think about the two functionally significantly different cases separately from each other as their implementations aren't in intermingled lines of code. And the style of putting the special cases / error checks first and using early returns is very common and idiomatic.

Reply:
The special case is size 0 is an optimization, no need to mmap and memcpy though it would be correct to do it, maybe we could even do it because it won't cost much.

The goal was to avoid duplicating the code that closes the file and frees the filename.